### PR TITLE
Localization

### PR DESCRIPTION
**Pull request by adamsitnik**
- [x] make LocalizationResources internal and static
- [x] tests
- [x] merge translation contributions (#1979, #1381)
- [x] move SDK [translations](https://github.com/dotnet/sdk/tree/main/src/Cli/dotnet/CommandLineValidation/xlf) here

I've not configured [OneLocBuild](https://github.com/dotnet/arcade/blob/main/Documentation/OneLocBuild.md) as we may move the code to dotnet/runtime where it's already configured for the whole repo.

To move the existing translations from SDK repo to here I've wrote a console app that took care of mapping the translations by ID, by original English text (some IDs has changed but text remained the same).

The app is ugly but it does what it's needed:

<details>

```cs
using System.Text;
using System.Transactions;
using System.Xml;
using System.Xml.Linq;

namespace CopyTranslations
{
    internal class Program
    {
        static void Main(string[] args)
        {
            DirectoryInfo sdkFolder = new DirectoryInfo(args[0]);
            DirectoryInfo sclFolder = new DirectoryInfo(args[1]);

            foreach (FileInfo sourceFile in sdkFolder.GetFiles("*.xlf"))
            {
                string extension = sourceFile.Name.Substring(sourceFile.Name.IndexOf('.'));
                string destinationFilePath = Path.Combine(sclFolder.FullName, $"Resources{extension}");

                using FileStream src = sourceFile.OpenRead();
                using XmlReader srcXml = XmlReader.Create(src);

                XmlDocument sourceDocument = new ();
                sourceDocument.Load(srcXml);

                XmlNode srcBodyNode = sourceDocument.ChildNodes[1]!.ChildNodes[0]!.ChildNodes[0]!;
                Dictionary<string, string> idToTarget = new ();
                Dictionary<string, string> sourceToTarget = new ();

                foreach (XmlNode node in srcBodyNode.ChildNodes)
                {
                    if (GetSource(node) != GetTarget(node)) // some has no translations
                    {
                        idToTarget.Add(GetId(node), GetTarget(node));
                        sourceToTarget.TryAdd(Normalize(GetSource(node)), GetTarget(node));
                    }
                }

                XmlDocument destinationDocument = new ();
                destinationDocument.Load(destinationFilePath);

                XmlNode dstBodyNode = destinationDocument.ChildNodes[1]!.ChildNodes[0]!.ChildNodes[0]!;

                foreach (XmlElement destinationElement in dstBodyNode.ChildNodes)
                {
                    string id = GetId(destinationElement);
                    string source = GetSource(destinationElement);

                    if (idToTarget.TryGetValue(id, out string? translation) // same IDs
                        || sourceToTarget.TryGetValue(Normalize(source), out translation)) // same English text
                    {
                        if (source.Contains("'{0}'") && translation.Contains("„{0}”"))
                        {
                            translation = translation.Replace("„{0}”", "'{0}'");
                        }
                        else if (source.Contains("'{0}'") && translation.Contains("\"{0}\""))
                        {
                            translation = translation.Replace("\"{0}\"", "'{0}'");
                        }
                        else if (source.Contains("'{0}'") && !translation.Contains("'{0}'"))
                        {
                            translation = translation.Replace("{0}", "'{0}'");
                        }

                        if (source.EndsWith('.') && !translation.EndsWith('.'))
                        {
                            translation += ".";
                        }

                        translation = translation.Replace(" :", ":");

                        destinationElement.ChildNodes[1]!.InnerText = translation;
                        destinationElement.ChildNodes[1]!.Attributes!["state"]!.InnerText = "translated";
                    }
                }

                destinationDocument.Save(destinationFilePath);
            }

            static string GetId(XmlNode node) => node.Attributes!["id"]!.InnerText;

            static string GetSource(XmlNode node) => node.ChildNodes[0]!.InnerText!;

            static string GetTarget(XmlNode node) => node.ChildNodes[1]!.InnerText!;

            static string Normalize(string formatString)
            {
                StringBuilder buffer = new(formatString);
                for (int i = 0; i < 4; i++)
                {
                    buffer.Replace("'{" + i + "}'", "{" + i + "}");
                }

                buffer.Replace(".", "");
                buffer.Replace(",", "");

                return buffer.ToString();
            }
        }
    }
}
```

</details>

### CODE REVIEW

File: src/System.CommandLine.Tests/System.CommandLine.Tests.csproj [hunk 1]
```diff
@@ -32,6 +32,8 @@
     <Compile Include="..\Common\ArgumentBuilder.cs" Link="Utility\ArgumentBuilder.cs" />
     <Compile Include="..\Common\OptionBuilder.cs" Link="Utility\OptionBuilder.cs" />
     <Compile Include="..\System.CommandLine.Suggest\DotnetMuxer.cs" Link="Utility\DotnetMuxer.cs" />
+    <Compile Include="..\System.CommandLine\LocalizationResources.cs" Link="LocalizationResources.cs" />
```
Comment (adamsitnik) [inline src/System.CommandLine.Tests/System.CommandLine.Tests.csproj:35]: I am including these types and the resources here so the tests can call the internal APIs to verify error messages.

File: src/System.CommandLine.ApiCompatibility.Tests/LocalizationTests.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+﻿using System.Globalization;
+using System.Linq;
+using Xunit;
+
+namespace System.CommandLine.ApiCompatibility.Tests
```
Comment (adamsitnik) [inline src/System.CommandLine.ApiCompatibility.Tests/LocalizationTests.cs:5]: I've added this type to a different project than `System.CommandLine.Tests` to ensure that it does not use the resources included in that project.

File: src/System.CommandLine/LocalizationResources.cs [hunk 3]
```diff
@@ -11,42 +11,28 @@ namespace System.CommandLine
...
         /// <summary>
         ///   Interpolates values into a localized string similar to Command &apos;{0}&apos; expects a single argument but {1} were provided.
         /// </summary>
-        public virtual string ExpectsOneArgument(SymbolResult symbolResult) => 
-            symbolResult is CommandResult
-                    ? GetResourceString(Properties.Resources.CommandExpectsOneArgument, symbolResult.Token().Value, symbolResult.Tokens.Count)
-                    : GetResourceString(Properties.Resources.OptionExpectsOneArgument, symbolResult.Token().Value, symbolResult.Tokens.Count);
+        internal static string ExpectsOneArgument(SymbolResult symbolResult) =>
+            symbolResult is CommandResult commandResult
+                    ? GetResourceString(Properties.Resources.CommandExpectsOneArgument, commandResult.Token.Value, symbolResult.Tokens.Count)
```
[the remark this answers is not in the document]
Comment (adamsitnik) [reply src/System.CommandLine/LocalizationResources.cs]: Great catch, fixed

**Comment by adamsitnik** [comment]
@KalleOlaviNiemitalo  I've removed all the unused methods and resources, thank you for the feedback!

I was also curious if it's not a bug that we don't use them. It turned out that we are now just using different methods for the same purposes:

https://github.com/dotnet/command-line-api/blob/b8a97611020a6b6c10710ad35517fab18f106b5c/src/System.CommandLine/LocalizationResources.cs#L55

https://github.com/dotnet/command-line-api/blob/b8a97611020a6b6c10710ad35517fab18f106b5c/src/System.CommandLine/LocalizationResources.cs#L19

**Review by jonsequitur — APPROVED** [review]

**Pull request merged**